File: ruby/perfect-numbers/perfect_numbers.rb
module PerfectNumber
  extend self

  def classify(number)
    raise RuntimeError if number <= 0
    case aliquot(number)
    when number then 'perfect'
    when ->(x) { x < number } then 'deficient'
    else 'abundant' 
    end
  end

  def aliquot(number)
    (1..number / 2).select { |divisor| (number % divisor).zero? }.sum
  end

end

File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number <= 0:
        raise ValueError("Classification is only possible for positive integers.")
    aliquot = []
    for i in range(1, number):
        if number % i == 0:
            aliquot.append(i)
    if sum(aliquot) == number:
        return "perfect"
    elif sum(aliquot) > number:
        return "abundant"
    else:
        return "deficient"

            
